log/logger.go

```diff
@@ -117,10 +117,11 @@ func WithInstrumentationVersion(version string) LoggerOption {
 // WithInstrumentationAttributes returns a [LoggerOption] that sets the
 // instrumentation attributes of a [Logger].
 //
-// The passed attributes will be de-duplicated.
+// Note that the passed attributes are de-duplicated by WithInstrumentationAttributes.
 func WithInstrumentationAttributes(attr ...attribute.KeyValue) LoggerOption {
```

Comment:
For discussion only (no need to adjust according to this before any conclusion): the following approach can resolve concurrency safety issues caused by external changes.

```go
func WithInstrumentationAttributes(attr ...attribute.KeyValue) LoggerOption {
	attrs := make([]attribute.KeyValue, len(attr))
	copy(attrs, attr)
	set := attribute.NewSet(attrs...)
	return loggerOptionFunc(func(config LoggerConfig) LoggerConfig {
		config.attrs = set
		return config
	})
}
```

Reply:
Also started discussing this above:

>> Could we clone the attributes? This may affect performance of applications that call WithInstrumentationAttributes in a hot code path
>
> I waas considering it as well and I do not see it as a good idea. It increases the performance overhead on the hot path. It does not fix all possible issues and it is also not needed in many scenarios,

@pellared when you said "does not fix all possible issues" were you referring to other options, or just WithInstrumentationAttributes? FWIW, as a user I think it would be better to be safe and take the hit to performance, once https://github.com/open-telemetry/opentelemetry-go/pull/7287 is available.

Reply:
Seems reasonable to me given https://github.com/open-telemetry/opentelemetry-go/pull/7287 is shipped

Reply:
Related comment by @dashpole https://github.com/open-telemetry/opentelemetry-go/pull/7287#pullrequestreview-3176986926:

> Should we deprecate `WithInstrumentationAttributes` and point users toward `WithInstrumentationAttributeSet`? As a user it isn't clear which I should use, and this seems to be a safer+better option.

Reply:
> when you said "does not fix all possible issues" were you referring to other options, or just WithInstrumentationAttributes? 

I meant `WithInstrumentationAttributes` without  https://github.com/open-telemetry/opentelemetry-go/pull/7287 being available. Then reusing attributes slices would not make any sense.

> FWIW, as a user I think it would be better to be safe and take the hit to performance, once https://github.com/open-telemetry/opentelemetry-go/pull/7287 is available.

This is questionable. However, given the precedence in https://github.com/open-telemetry/opentelemetry-go/blob/5e3b939f4bc5d33e202dd6217043b8b5391043c1/metric/instrument.go#L372C6-L376 and if https://github.com/open-telemetry/opentelemetry-go/pull/7287 is merged then I am fine with cloning.

Reply:
FYI https://github.com/open-telemetry/opentelemetry-go/pull/7287#pullrequestreview-3181379062

Reply:
@pellared sounds like a sensible change, but can we do that in another PR?

Reply:
> @pellared sounds like a sensible change, but can we do that in another PR?

Already done and merged :wink: